hazelcast/src/main/java/com/hazelcast/map/impl/EntryViews.java

```diff
@@ -58,11 +58,23 @@ public static <K, V> WanMapEntryView<K, V> createWanEntryView(Data key, Data val
                 .withVersion(record.getVersion())
                 .withHits(record.getHits())
                 .withLastAccessTime(record.getLastAccessTime())
-                .withLastUpdateTime(record.getLastUpdateTime())
+                .withLastUpdateTime(calculateLastUpdateTime(record, expiryMetadata))
                 .withCreationTime(record.getCreationTime())
                 .withLastStoredTime(record.getLastStoredTime())
                 .withTtl(expiryMetadata.getTtl())
                 .withMaxIdle(expiryMetadata.getMaxIdle())
                 .withExpirationTime(expiryMetadata.getExpirationTime());
     }
+
+    private static <V> long calculateLastUpdateTime(Record<V> record, ExpiryMetadata expiryMetadata) {
+        if (record.getLastUpdateTime() != Record.UNSET) {
+            return record.getLastUpdateTime();
+        }
+
+        if (expiryMetadata != ExpiryMetadata.NULL) {
+            return expiryMetadata.getLastUpdateTime();
+        }
+
+        return Record.UNSET;
+    }
```

Comment:
I think we can do this simplification for the backports for versions < 5.2?

```suggestion
    private static <V> long calculateLastUpdateTime(Record<V> record, ExpiryMetadata expiryMetadata) {
       if (expiryMetadata != ExpiryMetadata.NULL) {
            return expiryMetadata.getLastUpdateTime();
        }
        
        return record.getLastUpdateTime();
    }
}
```

Reply:
https://github.com/hazelcast/hazelcast/pull/23323/commits/985b45373ad1a38e6f892f4d14d1f31fbfb68e0a